NICN1.py

```diff
@@ -0,0 +1,283 @@
+        self.sequence = sequence.upper()
+        self.encoding = encoding
+        self._residue_names = {
+            "ALA": "A",
+            "ARG": "R",
+            "ASN": "N",
+            "ASP": "D",
+            "CYS": "C",
+            "GLN": "Q",
+            "GLU": "E",
+            "GLY": "G",
+            "HIS": "H",
+            "ILE": "I",
+            "LEU": "L",
+            "LYS": "K",
+            "MET": "M",
+            "PHE": "F",
+            "PRO": "P",
+            "SER": "S",
+            "THR": "T",
+            "TRP": "W",
+            "TYR": "Y",
+            "VAL": "V",
+        }
+        self._residue_mass = {
+            "A": 89,
+            "R": 174,
+            "N": 132,
+            "D": 133,
+            "C": 121,
+            "Q": 146,
+            "E": 147,
+            "G": 75,
+            "H": 155,
+            "I": 131,
+            "L": 131,
+            "K": 146,
+            "M": 149,
+            "F": 165,
+            "P": 115,
+            "S": 105,
+            "T": 119,
+            "W": 204,
+            "Y": 181,
+            "V": 117,
+        }
+
+    def _reformat_based_on_encoding(self) -> Union[str, List[str]]:
+        """Reformat the sequence based on the encoding."""
+        if self.encoding == 1:
+            return self.sequence
+        elif self.encoding == 3:
+            amino_acid_list = [
+                self.sequence[letter: letter + 3]
+                for letter in range(0, len(self.sequence), 3)
+            ]
+            return amino_acid_list
+        else:
+            raise IncorrectAminoacidEncodingError(
+                f"{self.encoding}-letter encoding is unavailable. "
+                f"Please, use 1 or 3 letter encoding "
+            )
+
+    def _make_one_letter(self) -> str:
+        """Convert the sequence to one-letter code."""
+        if self.encoding == 3:
+            one_letter_seq = str()
+            for aminoacid in self._reformat_based_on_encoding():
+                one_letter_seq += self._residue_names[aminoacid]
+            return one_letter_seq
+        return self.sequence
+
+    def __len__(self) -> int:
+        return len(self._reformat_based_on_encoding())
+
+    def __getitem__(self, index: int) -> Union[str, List[str]]:
+        if self.encoding == 3:
+            return "".join(self._reformat_based_on_encoding()[index])
+        return self.sequence[index]
+
+    def __str__(self) -> str:
+        return str(self.sequence)
+
+    def check_alphabet(self) -> bool:
+        """Check if the sequence contains valid amino acids."""
+        aminoacids = set(self._reformat_based_on_encoding())
+        for aminoacid in aminoacids:
+            if (
+                aminoacid not in self._residue_names.keys()
+                and aminoacid not in self._residue_names.values()
+            ):
+                raise IncorrectAminoacidError(
+                    f"{aminoacid} is not a supported amino acid!"
+                )
```

Comment:
Мне кажется, что проще было бы сделать через множества
```suggestion
        
          if (
              set(aminoacids) <= set(self._residue_names.keys())
              and set(aminoacids) <= set(self._residue_names.values())
          ):
              raise IncorrectAminoacidError(
                  f"{aminoacid} is not a supported amino acid!"
              )
```